misk/src/main/kotlin/misk/logging/LogRateLimiter.kt

```diff
@@ -0,0 +1,56 @@
+package misk.logging
+
+import misk.web.interceptors.LogRequestResponse
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Holds rate limiters for logging success and error responses. There is a rate limiter for every
+ * action and service that calls it. The rate limiters are created according to the value set in
+ * [LogRequestResponse] annotation, which is the number of buckets per pod per service.
+ */
+@Singleton
+class LogRateLimiter @Inject constructor(
+  private val rateLimiterFactory: RateLimiter.Factory
+) {
+  private val liveConfigRef =
+    AtomicReference(LiveConfig(
+      rateLimiters = ConcurrentHashMap<LogBucketId, RateLimiter>(),
+      errorRateLimiters = ConcurrentHashMap<LogBucketId, RateLimiter>()
+    ))
+
+  fun tryAcquireOnError(bucketId: LogBucketId, permits: Long): Boolean {
```

Comment:
Why is `permits` an argument? You return a boolean here, so why would you ever acquire more than 1?

I think there might be performance tradeoffs w.r.t. acquiring a batch of permits, but I don't think that's relevant here. cc: @swankjesse

Reply:
I had passed in more than one permit in LogRateLimiterTest, but I agree for logging we should only ever tryAcquire on 1 bucket